docs/plots/backends.rst

```diff
@@ -0,0 +1,389 @@
+classes that connect the Sherpa plot objects with the plotting backend by
+translating Sherpa options into backend-specific commands. The details of that
+are explained below, but, in short, a backend class would translate the sherpa
+backend-independent "plot a line in red from here to there" into backend
+specific ``plt.plot(x,y, linecolor='r')`` (if the backend is :term:`matplotlib`).
+
+Instead of utilizing Sherpa classes and commands to plot data, users can
+alternatively just access the data in the Sherpa data objects (e.g. the x and y
+values of a dataset) and perform plotting operations directly with any plotting
+backend that is available to them. This method may be less convenient, but it
+works for plotting backends not (yet) supported by Sherpa.
+
+Which backend is used?
+======================
+
+When the `sherpa.plot` module is first imported, Sherpa tries to import the
+backends installed with Sherpa in the order listed in the
+``options.plot_pkg`` setting from the ``sherpa.rc`` startup file.
+The first module that imports successfully is set as the active
+backend. The following command prints the name and the location
+on disk of that module::
+
+   >>> from sherpa import plot
+   >>> print(plot.backend.name)
+
+.. warning::
+    Of course, you could be tempted to write::
+
+      >>> from sherpa.plot import backend
+      >>> print(backend.name)
+
+    However, ``backend`` is now a reference to the backend that
+    was active when the import was done. If the backend is changed
+    later, ``backend`` will still refer to the old backend. Thus,
+    always use ``plot.backend`` to access the active backend.
+
+Change the backend
+------------------
+
+.. warning::
+   When the plotting backend is changed, new plotting objects must be
+   initialized because objects like e.g. `sherpa.plot.DataPlot` keep a
+   copy of all default parameters of the backend that was active when they
+   are initialized. A different backend might not understand all those
+   parameters.
+
+   In particular, Sherpa's UI layer (which contains functions such as
+   `sherpa.astro.ui.plot_data`) keeps a reference to specific plot objects.
+   Changing the backend by itself will not change those references, and thus
+   those function might not work with a new backend.
+
+After the initial import, the backend can be changed by loading one of
+the plotting backends shipped with sherpa (or any other module that
+provides the same interface)::
+
+  >>> from sherpa.plot import set_backend
+  >>> set_backend('pylab')
+
+Sherpa also provides a context manager to change the backend for just one plot::
+
+  >>> from sherpa.plot import TemporaryPlottingBackend
+  >>> with TemporarypPlottingBackend('pylab'):
+  ...     x = [1,2,3,4]
+  ...     # do some plotting with x
+
+The list of available plotting backend names and the classes that implement them is::
+
+  >>> from sherpa.plot.backends import PLOT_BACKENDS
+  >>> print(PLOT_BACKENDS)
```

Comment:
Would it be better to acknowledge that `PLOT_BACKENDS` is a dict-like object and instead say the following?

```suggestion
  >>> print(PLOT_BACKENDS.keys())
```